### DISCUSSION TITLE
什么时候支持文本框取消翻译

### DISCUSSION THREAD

**Discussion by vipcxj** [immersive-translate/immersive-translate#2241] [Ideas]
### 新功能 | New Feature

1. 支持撤销文本框翻译

### 为什么您想要添加这些功能 | Why you want to add these features

1. 没有此功能的软件就是个shit，不容反驳。误翻译直接把原文给弄没了，ctrl-z都没用，等于严重bug，不能撤销那么这个应用就是个shit。或许已经有该功能，但教程里找不到。

**Comment by jasperyou** [comment]
请提供具体示例，并附上可重现的步骤和版本号。目前我在文本框中输入单词时，发现该单词无法被翻译，并且没有撤销选项。如果需要取消当前网页的翻译结果，可以再次点击按钮，绿色标识消失后即可恢复原状态。

![1730690059113](https://github.com/user-attachments/assets/62c36681-f4e9-4225-ae16-277f1ceb34c3)

**Reply by vipcxj** [reply]
你这个按钮又不是取消文本框翻译的，文本框翻译指那3下空格，一锤子买卖，用了就回不去，ctrl-z直接清空文本框

**Reply by vipcxj** [reply]
刚试了下github ctrl-z能撤销到翻译前，但github可以不代表其他网站可以，这个撤销应该是github自带的，反正我其他网站直接寄了。这个功能需要沉浸式翻译直接支持，毕竟只有插件百分百记得翻译前的原文

**Reply by jasperyou** [reply]
> 刚试了下github ctrl-z能撤销到翻译前，但github可以不代表其他网站可以，这个撤销应该是github自带的，反正我其他网站直接寄了。这个功能需要沉浸式翻译直接支持，毕竟只有插件百分百记得翻译前的原文

你方便录个视频吗? 讲不清具体那个网站不行，Google，Github都可以。

**Reply by jasperyou** [reply]
这个功能叫输入框增强，在HTML标签里输入框与文本框是不一样的，目前看来输入框是没有问题的。

![1730699720031](https://github.com/user-attachments/assets/9332d5c9-c565-4d31-b476-9590a687883a)

**Reply by vipcxj** [reply]
知乎

**Reply by vipcxj** [reply]
在知乎随便发个评论，不需要真的发出去，然后点三下空格，中文变成了英文，然后怎么变回来？ctrl-z直接清空输入。另外我总觉得你没理解我的意思。你截图的google，不要说撤销翻译了，压根不支持点三下空格翻译。不过这个和本贴也没什么关系

**Reply by jasperyou** [reply]
> 在知乎随便发个评论，不需要真的发出去，然后点三下空格，中文变成了英文，然后怎么变回来？ctrl-z直接清空输入。另外我总觉得你没理解我的意思。你截图的google，不要说撤销翻译了，压根不支持点三下空格翻译。不过这个和本贴也没什么关系

1.知乎的评论框不是严格意义上的文本框，文本框与输入框都是支持对内容撤回的，不需要插件支持。
2.Google是支持三下空格翻译的，请查看文档：chrome-extension://bpoadfkcbjbfhfodiogcnhhhpibjhbnh/options.html#input
3.这里是开源社区，欢迎继续补充问题细节，当然你也可以尝试解决这个看似小众的问题。

![1730707069012](https://github.com/user-attachments/assets/111cacea-0de5-4eb3-bb29-af5bef06e6d0)

**Reply by vipcxj** [reply]
> > 在知乎随便发个评论，不需要真的发出去，然后点三下空格，中文变成了英文，然后怎么变回来？ctrl-z直接清空输入。另外我总觉得你没理解我的意思。你截图的google，不要说撤销翻译了，压根不支持点三下空格翻译。不过这个和本贴也没什么关系
> 
> 1.知乎的评论框不是严格意义上的文本框，文本框与输入框都是支持对内容撤回的，不需要插件支持。 2.Google是支持三下空格翻译的，请查看文档：chrome-extension://bpoadfkcbjbfhfodiogcnhhhpibjhbnh/options.html#input 3.这里是开源社区，欢迎继续补充问题细节，当然你也可以尝试解决这个看似小众的问题。
> 
> ![1730707069012](https://private-user-images.githubusercontent.com/33439399/382678726-111cacea-0de5-4eb3-bb29-af5bef06e6d0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA3MTIyMDgsIm5iZiI6MTczMDcxMTkwOCwicGF0aCI6Ii8zMzQzOTM5OS8zODI2Nzg3MjYtMTExY2FjZWEtMGRlNS00ZWIzLWJiMjktYWY1YmVmMDZlNmQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA0VDA5MTgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMyOWMxMDdhNThhZTE3NDc1YzA1OTYzNmE2YjViZTllMzQ4NDg5OWFjZTdhMDFkYjZjZWE1MmU4ODJhMDBkNDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.oHkRtX6crz4lsu1Mq6KUOfk2ym__lzLmIW80bP3eLXg)

我不觉得这是小众的问题，既然负责翻译，自然要负责撤销，把责任丢给网页本身算什么逻辑。虽然是开源社区，但这款产品只有付费vip的，并不是完全免费的，至少我是付费用户，体验有问题还要我自己写代码解决吗？

**Reply by vipcxj** [reply]
如果我真自己写代码解决了，你会把利润分给我吗？
